src/gluonts/model/seq2seq/_forking_network.py

```diff
@@ -66,26 +67,36 @@ def __init__(
 class ForkingSeq2SeqTrainingNetwork(ForkingSeq2SeqNetworkBase):
     # noinspection PyMethodOverriding
     def hybrid_forward(
-        self, F, past_target: Tensor, future_target: Tensor
+        self,
+        F,
+        past_target: Tensor,
+        future_target: Tensor,
+        past_observed_values: Tensor,
+        future_observed_values: Tensor,
     ) -> Tensor:
         """
         Parameters
         ----------
         F: mx.symbol or mx.ndarray
             Gluon function space
         past_target: Tensor
-            FIXME
+            shape (num_ts, encoder_length, 1)
         future_target: Tensor
-            shape (num_ts, encoder_length, 1) FIXME
+            shape (num_ts, encoder_length, decoder_length)
+        past_observed_values: Tensor
+            shape (num_ts, encoder_length, 1)
+        future_observed_values: Tensor
+            shape (num_ts, encoder_length, decoder_length)
 
         Returns
         -------
-        loss with shape (FIXME, FIXME)
+        loss with shape (num_ts, decoder_length)
         """
 
         # FIXME: can we factor out a common prefix in the base network?
+
         feat_static_real = F.zeros(shape=(1,))
-        past_feat_dynamic_real = F.zeros(shape=(1,))
+        past_feat_dynamic_real = past_observed_values
```

Comment:
Why is `past_feat_dynamic_real` set to  `past_observed_values`? Shouldn't `past_observed_values` be used for masking, it's not a covariate itself is it?

Reply:
Since `past_target` is also an input feature to the network and we are masking some entries in it, network should be able to distinguish between the actual values and the masked values. Passing `past_observed_values` as a feature would allow the network to make that distinction and possibly ignore the masked values.